docs/reference/filebeat/filebeat-input-azure-eventhub.md

```diff
@@ -42,12 +61,108 @@ The name of the eventhub users would like to read from, field required.
+
+### Authentication Types
+
+The following authentication types are supported:
+
+- **`connection_string`** (default if `auth_type` is not specified): Uses Azure Event Hubs and Storage Account connection strings
+- **`client_secret`**: Uses Azure Active Directory service principal with client secret credentials
+
+### Required Permissions
+
+When using `client_secret` authentication, the service principal needs the following Azure RBAC permissions:
+
+**For Azure Event Hubs:**
+- `Azure Event Hubs Data Receiver` role on the Event Hubs namespace or Event Hub
+- Alternatively, a custom role with the following permissions:
+  - `Microsoft.EventHub/namespaces/eventhubs/read`
+  - `Microsoft.EventHub/namespaces/eventhubs/consumergroups/read`
+
+**For Azure Storage Account:**
+- `Storage Blob Data Contributor` role on the Storage Account or container
+- Alternatively, a custom role with the following permissions:
+  - `Microsoft.Storage/storageAccounts/blobServices/containers/read`
+  - `Microsoft.Storage/storageAccounts/blobServices/containers/write`
+  - `Microsoft.Storage/storageAccounts/blobServices/containers/delete`
+  - `Microsoft.Storage/storageAccounts/blobServices/generateUserDelegationKey/action`
+
+For detailed instructions on how to set up an Azure AD service principal and configure permissions, refer to the official Microsoft documentation:
+
+- [Create an Azure service principal with Azure CLI](https://learn.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli)
+- [Create an Azure AD app registration using the Azure portal](https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+- [Assign Azure roles using Azure CLI](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-cli)
+- [Azure Event Hubs authentication and authorization](https://learn.microsoft.com/en-us/azure/event-hubs/authorize-access-azure-active-directory)
+- [Authorize access to blobs using Azure Active Directory](https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory)
+
 ## `connection_string` [_connection_string]
 
-The connection string required to communicate with Event Hubs, steps here [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string).
+The connection string required to communicate with Event Hubs when using `connection_string` authentication, steps here [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-get-connection-string).
+
+Required when `auth_type` is set to `connection_string` or when `auth_type` is not specified (defaults to `connection_string` for backwards compatibility).
```

Comment:
@colleenmcginnis Do I add {applies_to}`stack: ga 9.3.0` right above this section?